lib/kubeclient/common.rb

```diff
@@ -233,6 +233,27 @@ def create_rest_client(path = nil)
       RestClient::Resource.new(@api_endpoint.merge(path).to_s, options)
     end
 
+    def do_rest_request(args)
```

Comment:
This one looks really similar to create_rest_client. Is there anyway to reuse it?

Reply:
Extracting the common options should be enough actually (`common_options`?)

Also consider keeping the pattern we have in this file of passing path and headers later on and memoize the client if possible.

```
rest_client[ns_prefix + resource_name]
  .get({ 'params' => params }.merge(@headers))
end
```

Reply:
can probably switch to using `do_rest_request` everywhere.
except `rest_client`, `create_rest_client` are public, in theory someone might depend on them.

another direction: you only did this to be able to pass `:block_response`, right?
I think RestClient::Resource can pass through any option, including `:block_response`...
https://github.com/rest-client/rest-client/blob/master/lib/restclient/resource.rb#L51

Reply:
(to clarify, I don't mind the textual duplication, but I'd rather avoid going through 2 code paths in RestClient.)

Reply:
@cben I agree that 2 code paths are bad. Unfortunately, you can't pass `:block_response` to `RestClient.get`, et al. (see https://github.com/rest-client/rest-client#block_response-receives-raw-nethttpresponse), so all the other code paths would have to be refactored to use `RestClient::Request.execute` rather than `create_rest_client` as they currently do. Since the scope of this change would be pretty large, I didn't want to include it in my PR.

Reply:
I see.  Agreed, rewriting everything to use `execute` is riskier change, might eventually be worth it, but not for this PR.
:+1: